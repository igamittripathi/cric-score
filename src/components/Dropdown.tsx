import React from 'react';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';

interface IDropdown {
    onChange?: (event: any) => void;
    value?: string | number;
    id?: string;
    labelId?: string;
    options: Array<string | number>
}


const Dropdwon = ({ onChange, value, id, labelId, options }: IDropdown) => (
    <div>
        <InputLabel id="select-label"> Type </InputLabel>
        <Select
            labelId={labelId}
            id={id}
            value={value}
            onChange={onChange}
        >
            {options.map((item: string | number, idx: number) => <MenuItem key={idx} value={item}>{item}</MenuItem>)}

        </Select>
    </div>
)

export default Dropdwon;
