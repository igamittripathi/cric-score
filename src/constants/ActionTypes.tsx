export const GET_TEAMS = 'GET_TEAMS'
export const SAVE_TEAMS_INIT = 'SAVE_TEAMS_INIT'
export const UPDATE_TEAMS_INIT = 'UPDATE_TEAMS_INIT';
export const SAVE_TEAM_SUCCESS= 'SAVE_TEAM_SUCCESS';
