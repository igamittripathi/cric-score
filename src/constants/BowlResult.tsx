export type BowlResultType = 0|1|2|4|5|6|'No Bowl'|'Wicket'|'Wide Bowl';

export const BowlResults = [0,1,2,4,5,6,'No Bowl','Wicket','Wide Bowl'];
