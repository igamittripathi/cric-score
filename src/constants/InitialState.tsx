import { ITeams } from "../interfaces";

export const initialState: ITeams  = {
    team_a: {
        name: 'India',
        playres: [],
        id: Math.floor(Math.random() * 100),
        totalRun: 0,
        overs: [],
        isInningCompleted: false
    },
    team_b: {
        name: 'England',
        playres: [],
        id: Math.floor(Math.random() * 100),
        totalRun: 0,
        overs: [],
        isInningCompleted: false
    } 
}
