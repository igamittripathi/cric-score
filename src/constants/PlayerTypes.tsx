export const PlayerTypes: string[] = ['batsman', 'bowler']
