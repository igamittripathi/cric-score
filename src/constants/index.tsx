export * from './PlayerTypes';
export * from './BowlResult';
export * from './InitialState';
export * from './ActionTypes';
