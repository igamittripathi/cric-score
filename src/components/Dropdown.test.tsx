import * as React from 'react';
import { Dropdwon } from './Dropdown';
import { shallow, mount } from "enzyme";
import { FormControl, InputLabel, Select, MenuItem } from '@material-ui/core';
import { BowlResults } from '../constants';


describe("Dropdwon", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(<Dropdwon options={BowlResults} value={0} />);
    });

    test("Cricle should render properly", () => {
        wrapper = shallow(<Dropdwon />);
        expect(wrapper).toMatchSnapshot();
    });

    test("Dropdwon should render FormControl", () => {
        expect(wrapper.find(FormControl)).toHaveLength(1);
    });

    test("Dropdwon should render InputLabel", () => {
        expect(wrapper.find(InputLabel)).toHaveLength(1);
    });

    test("Dropdwon should render Select", () => {
        expect(wrapper.find(Select)).toHaveLength(1);
    });

    test("Dropdwon should render MenuItem", () => {
        expect(wrapper.find(MenuItem)).toBeDefined();
    });

    test("allows us to set props", () => {
        expect(wrapper.props().value).not.toEqual(1);
        wrapper.setProps({ value: 6 });
        expect(wrapper.props().value).toEqual(6);
    });

});