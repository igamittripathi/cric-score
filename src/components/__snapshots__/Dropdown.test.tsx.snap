// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdwon Cricle should render properly 1`] = `
<WithStyles(ForwardRef(FormControl))
  className="makeStyles-margin-1"
>
  <WithStyles(ForwardRef(InputLabel))
    error={false}
    htmlFor="selectundefined"
    required={false}
  >
    Type
  </WithStyles(ForwardRef(InputLabel))>
  <WithStyles(ForwardRef(Select))
    error={false}
    input={<ForwardRef(WithStyles) />}
    required={false}
  />
</WithStyles(ForwardRef(FormControl))>
`;
